gtda/images/filtrations.py

```diff
@@ -349,3 +350,135 @@ def transform(self, X, y=None):
         Xt = np.concatenate(Xt)
 
         return Xt
+
+
+@adapt_fit_transform_docs
+class DilationFiltration(BaseEstimator, TransformerMixin):
+    """Filtrations of 2D/3D binary images based on distances to activated
+    pixels.
+
+    The dilation filtration assigns to each deactivated pixel of an image a
+    pixel value corresponding to the Manhattan distance between the pixel and
+    its closest activated pixel. The dilation process is iterative and the
+    number of iterations `n_iterations` sets the maximum distance considered.
+    Deactivated pixels further away from activated pixels are assigned the
+    maximum Manhattan distance between two pixels on the image and activated
+    pixels are assigned a value of 0.
+
+    Parameters
+    ----------
+    n_iterations : int or None, optional, default: ``None``
+        Number of iterations in the dilation process. ``None`` means dilation
+        over the full image.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    n_iterations_ : int
+        Effective number of iterations in the dilation process. Set in
+        :meth:`fit`.
+
+    max_value_: float
+        Maximum pixel value among all pixels in all images of the collection.
+        Set in :meth:`fit`.
+
+    See also
+    --------
+    gtda.homology.CubicalPersistence, Binarizer
+
+    """
+    _hyperparameters = {'n_iterations_': [int, (1, np.inf)]}
+
+    def __init__(self, n_iterations=None, n_jobs=None):
+        self.n_iterations = n_iterations
+        self.n_jobs = n_jobs
+
+    def _calculate_dilation(self, X):
+        Xd = X * 1.
+
+        for iteration in range(1, min(self.n_iterations_,
+                                      self.max_value_) + 1):
+            Xtemp = np.asarray([ndi.binary_dilation(Xd[i])
+                                for i in range(Xd.shape[0])])
+            Xd += (Xd + Xtemp == 1) * Xtemp * (iteration + 1)
```

Comment:
A pair of parentheses in the first factor on the RHS would help readability here.

Reply:
Furthermore, if `(Xd + Xtemp)==1` contains only `False`, we _could_ terminate the `for` loop early. So for instance one could do something in the spirit of
```
Xnew = (Xd + Xtemp) == 1
if np.any(Xnew):
    Xd[Xnew] = iteration + 1
else:
    break
```